homeassistant/components/comfoconnect/__init__.py

```diff
@@ -48,21 +53,41 @@
 )
 
 
-def setup(hass, config):
-    """Set up the ComfoConnect bridge."""
+async def async_setup(hass, config):
+    """Set up the ComfoConnect integration."""
+    if hass.config_entries.async_entries(DOMAIN):
+        return True
+    if DOMAIN in config:
+        # import configuration using config flow
+        sensors = None
+        for sensor in config["sensor"]:
+            if sensor.get("platform") == DOMAIN:
+                sensors = sensor.get("resources")
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT, "import_sensors": sensors},
+                data=config[DOMAIN],
+            )
+        )
+
+    return True
 
-    conf = config[DOMAIN]
-    host = conf[CONF_HOST]
-    name = conf[CONF_NAME]
-    token = conf[CONF_TOKEN]
-    user_agent = conf[CONF_USER_AGENT]
-    pin = conf[CONF_PIN]
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Set up platform from a ConfigEntry."""
+    host = entry.data[CONF_HOST]
+    name = entry.data.get(CONF_NAME, DEFAULT_NAME)
+    token = entry.data.get(CONF_TOKEN, DEFAULT_TOKEN)
+    user_agent = entry.data.get(CONF_USER_AGENT, DEFAULT_USER_AGENT)
+    pin = entry.data.get(CONF_PIN, DEFAULT_PIN)
```

Comment:
The defaults should already be in the entry data (added by the config flow).
There is no need to resolve them here?